From 8273e50ae95918687aefe16851c6db894772f4b8 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@apache.org>
Date: Fri, 20 Mar 2015 17:03:03 -0700
Subject: [PATCH 2260/2748] HADOOP-11447. Add a more meaningful toString
 method to SampleStat and MutableStat. (kasha)

(cherry picked from commit fe5c23b670c773145b87fecfaf9191536e9f1c51)

Change-Id: I61cd3755a9e6fd7107efe53baf8eeee33a6318bb
---
 .../apache/hadoop/metrics2/lib/MutableStat.java    |    4 ++++
 .../apache/hadoop/metrics2/util/SampleStat.java    |   13 +++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableStat.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableStat.java
index fa1e33d..5108624 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableStat.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableStat.java
@@ -159,4 +159,8 @@ public void resetMinMax() {
     minMax.reset();
   }
 
+  @Override
+  public String toString() {
+    return lastStat().toString();
+  }
 }
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleStat.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleStat.java
index 589062a..cd9aaa4 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleStat.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleStat.java
@@ -137,6 +137,19 @@ public double max() {
     return minmax.max();
   }
 
+  @Override
+  public String toString() {
+    try {
+      return "Samples = " + numSamples() +
+          "  Min = " + min() +
+          "  Mean = " + mean() +
+          "  Std Dev = " + stddev() +
+          "  Max = " + max();
+    } catch (Throwable t) {
+      return super.toString();
+    }
+  }
+
   /**
    * Helper to keep running min/max
    */
-- 
1.7.9.5

